tests/url_utils.test.ts

```diff
@@ -60,20 +50,74 @@ describe("stateToUrlParamString", () => {
         "&feature=%D0%9F%D1%80%D0%B8%D0%B2%D0%B5%D1%82%2C%20%D0%BC%D0%B8%D1%80"
     );
   });
+});
 
-  it("Ignores bad track values", () => {
-    const result = stateToUrlParamString({ time: -56 });
-    expect(result).to.be.empty;
+describe("loadParamsFromUrlQueryString", () => {
+  it("Handles empty query strings", () => {
+    const result = loadParamsFromUrlQueryString("");
+    expect(result).to.deep.equal({});
+  });
+});
+
+describe("Loading + saving from URL query strings", () => {
+  it("Encodes + Decodes URL strings", () => {
+    const originalParams = {
+      collection: "https://some-url.com/collection.json", // https%3A%2F%2Fsome-url.com%2Fcollection.json
+      dataset: "你好世界", // %E4%BD%A0%E5%A5%BD%E4%B8%96%E7%95%8C
+      feature: "Привет, мир", // %D0%9F%D1%80%D0%B8%D0%B2%D0%B5%D1%82%2C%20%D0%BC%D0%B8%D1%80
+    };
+    const queryString = stateToUrlQueryString(originalParams);
+    const parsedParams = loadParamsFromUrlQueryString(queryString);
+    expect(parsedParams).deep.equals(originalParams);
   });
 
-  it("Gets URL parameters", () => {
-    const result = stateToUrlParamString({
+  it("Saves and retrieves URL params correctly", () => {
+    // This will need to be updated for any new URL params.
+    // The use of `Required` makes sure that we don't forget to update this test :)
+    const originalParams: Required<UrlParams> = {
       collection: "collection",
       dataset: "dataset",
       feature: "feature",
       track: 25,
       time: 14,
-    });
-    expect(result).to.equal("?collection=collection&dataset=dataset&feature=feature&track=25&t=14");
+      thresholds: [
+        { featureName: "f1", min: 0, max: 0 },
```

Comment:
is there a check somewhere that `max `is larger than `min`?

Reply:
The UI handles the case where the values are reversed, yes, but it's not directly handled in `url_utils`. I do think it makes sense for it to enforce min/max, though, so I'll make that change and add a test for it.